metriq_gym/process.py

```diff
@@ -0,0 +1,156 @@
+"""Benchmark processing and calculation utilities."""
+import json
+import math
+import statistics
+
+from scipy.stats import binom
+
+from qiskit_ibm_runtime import QiskitRuntimeService
+from qiskit.providers import Job
+
+from metriq_gym.bench import BenchJobResult, BenchProvider
+
+def get_job(result: BenchJobResult) -> Job:
+    if result.provider == BenchProvider.IBMQ:
+        return QiskitRuntimeService().job(result.id)
+
+    raise ValueError(f"Cannot find provider {job.provider}.")
+
+
+def get_job_result(job: Job, partial_result: BenchJobResult):
+    result = job.result()
+    partial_result.counts = result.get_counts()
+    partial_result.interval = result.time_taken
+
+    return partial_result
+
+
+def poll_job_results(jobs_file: str) -> [BenchJobResult]:
+    """Run quantum volume benchmark using QrackSimulator and return structured results.
+
+    Args:
+        jobs_file (str): Name of jobs file to check.
+
+    Returns:
+        An array of newly-completed BenchJobResult instances.
+    """
+
+    results = []    
+    lines_out = []
+    with open(jobs_file, 'r') as file:
+        lines = file.readlines()
+        for line in lines:
+            result = BenchJobResult(**(json.loads(line)))
+            job = get_job(result)
+            status = job.status()
+            if status == JobStatus.RUNNING:
+                # Still running
+                logging.info(f"Job still in-progress: " + job.job_id())
+                lines_out.append(line)
+            elif status == DONE:
+                # Success
+                logging.info(f"Job successful: " + job.job_id())
+                results.append(get_job_result(job, result))
+            else:
+                # Failure
+                logging.info(f"Job failed: " + job.job_id())
+
+    with open(jobs_file, 'w') as file:
+        file.writelines(lines_out)
+
+    return results
+
+
+def calc_trial_stats(ideal_probs: dict[int, float], counts: dict[str, int], interval: float, 
```

Comment:
Could the input to this function be a `BenchJobResult`?

Reply:
No, because now `BenchJobResult` has been modified to include results for _all_ requested _trials_ in the same "job" (saving significant overhead and complexity).